app/src/main/java/org/mozilla/fenix/tabstray/syncedtabs/SyncedTabs.kt

```diff
@@ -0,0 +1,339 @@
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.PathEffect
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import mozilla.components.browser.storage.sync.SyncedDeviceTabs
+import mozilla.components.browser.storage.sync.TabEntry
+import mozilla.components.browser.toolbar.MAX_URI_LENGTH
+import mozilla.components.feature.syncedtabs.view.SyncedTabsView
+import org.mozilla.fenix.NavGraphDirections
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.compose.SecondaryText
+import org.mozilla.fenix.ext.toShortUrl
+import org.mozilla.fenix.theme.FirefoxTheme
+import mozilla.components.browser.storage.sync.Tab as SyncTab
+
+/**
+ * Top-level list UI for displaying Synced Tabs in the Tabs Tray.
+ *
+ * @param syncedTabs [SyncedTabsListItem] The tab UI items to be displayed.
+ * @param listener [SyncedTabsView.Listener] to apply to this layout node.
+ */
+@Composable
+fun SyncedTabsList(syncedTabs: List<SyncedTabsListItem>, listener: SyncedTabsView.Listener) {
+    val listState = rememberLazyListState()
+    LazyColumn(
+        state = listState,
+        verticalArrangement = Arrangement.spacedBy(8.dp),
+        modifier = Modifier.fillMaxSize()
+    ) {
+        items(syncedTabs) { syncedTabItem ->
+            when (syncedTabItem) {
+                is SyncedTabsListItem.Device -> SyncedTabsDeviceItem(syncedTabItem.displayName)
+                is SyncedTabsListItem.Error -> SyncedTabsErrorItem(
+                    syncedTabItem.errorType,
+                    syncedTabItem.navController
+                )
+                is SyncedTabsListItem.NoTabs -> SyncedTabsNoTabsItem()
+                is SyncedTabsListItem.Tab -> SyncedTabsTabItem(
+                    syncedTabItem.tab.active().title,
+                    syncedTabItem.tab.active().url.toShortUrl(components.publicSuffixList).take(MAX_URI_LENGTH),
+                ) { listener.onTabClicked(syncedTabItem.tab) }
+            }
+        }
+    }
+}
+
+/**
+ * Text header for sections of synced tabs
+ *
+ * @param deviceName The name of the user's device connected that has synced tabs.
+ */
+@Composable
+fun SyncedTabsDeviceItem(deviceName: String) {
+    PrimaryText(
```

Comment:
nit: in the designs, we needed to add a horizontal line above the device name. I think it's fine to leave out if you're planning on picking up https://github.com/mozilla-mobile/fenix/issues/19942 next.

Reply:
Good catch! I noticed that without sticky headers it did become hard to tell one section from the other. My intent was to immediately do the sticky header issue next.